internal/admission/errors.go

```diff
@@ -14,6 +14,9 @@ const (
 	ErrTextPluginConfigValidationFailed       = "unable to validate plugin schema"
 	ErrTextPluginConfigViolatesSchema         = "plugin failed schema validation: %s"
 	ErrTextPluginSecretConfigUnretrievable    = "could not load secret plugin configuration"
+	ErrTextVaultConfigUnmarshalFailed         = "failed to unmarshal validate configurations: %v"
```

Comment:
I think this was maybe supposed to be "vault configuration" instead of "validate configuration"?

Should this be possible in practice anyway? Do we have any examples? For KongPlugin `config` (also an `apiextensionsv1.JSON` this sort of thing seems like it should be rejected by the API server before it even hits our webhook:

```
$ echo 'apiVersion: configuration.konghq.com/v1
kind: KongPlugin                                                    
metadata:       
  name: kong-id
config:        
  header_name: kong-id
  !!!                 
plugin: correlation-id' | kubectl apply -f -
error: error parsing STDIN: error converting YAML to JSON: yaml: line 8: could not find expected ':'
```
I guess we have it as a "technically we check it regardless because the Go unmarshal call can return an error, even though _we_ should never see it".